### PR TITLE
Replace {BatchPush,ClientView}Info with HttpRequestInfo

### PR DESCRIPTION
**Pull request by arv**
Towards #290

### CODE REVIEW

File: src/sync/types.rs [hunk 1]
```diff
@@ -1,22 +1,12 @@
...
-    pub http_status_code: u16,
-    #[serde(rename = "errorMessage")]
-    pub error_message: String,
-    // TODO BatchPushResponse BatchPushResponse `json:"batchPushResponse"`
-}
-
-#[derive(Clone, Debug, Default, Deserialize, PartialEq, Serialize)]
-pub struct ClientViewInfo {
+#[derive(Serialize, Deserialize)]
+#[cfg_attr(test, derive(Clone, Debug, PartialEq))]
```
Comment (arv) [inline src/sync/types.rs:8]: Not sure if is worth doing the cfg_attr dance? Seems slightly cleaner to have less derives.

**Comment by arv** [comment]
TBR

File: src/sync/types.rs (hunk 1 above)
Comment (phritz) [reply src/sync/types.rs:8]: It doesn't hurt, we should probably be more careful with this kind of thing in general. Suspect as aaron has said there will come a point in time when we find market fit that we do a big push to reduce size.

**Review by phritz — COMMENTED** [review]
LGTM

**Pull request merged**